verifier/src/x509.rs

```diff
@@ -6,5 +6,177 @@ mod certs;
 mod crl;
 mod error;
 
+extern crate alloc;
+use alloc::string::String;
+use alloc::vec;
 pub use error::Error;
 pub type Result<T> = core::result::Result<T, Error>;
+
+use mbedtls::{
+    alloc::List as MbedtlsList,
+    hash::Type as HashType,
+    pk::{EcGroupId, Type as PkType},
+    x509::{Certificate, Profile},
+};
+
+#[derive(Clone)]
+pub struct TrustAnchor(MbedtlsList<Certificate>);
```

Comment:
I'm not a fan of the  `MbedtlsList<Certificate>` type used for much of the interface, so chose to separate out the trustanchor form the actual certificatelist/chain